src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -30,7 +30,7 @@ internal partial class Binder
         // If we have no modifiers then the modifiers array is null; if we have any modifiers
         // then the modifiers array is non-null and not empty.
 
-        private Tuple<ImmutableArray<RefKind>, ImmutableArray<TypeSymbol>, ImmutableArray<string>, bool> AnalyzeAnonymousFunction(
+        private Tuple<ImmutableArray<RefKind>, ImmutableArray<TypeSymbol>, ImmutableArray<string>, ImmutableArray<bool>, bool> AnalyzeAnonymousFunction(
```

Comment:
If the index is meaningful then I think this should be an ImmutableArray of "modifier info" (a new struct type).